inc/agent.class.php

```diff
@@ -304,6 +331,11 @@ public function handleAgent($metadata) {
          $input['tag'] = $metadata['tag'];
       }
 
+      $input['ip_protocol'] = 'http';
+      if (!isset($metadata['httpd-plugins']['ssl'])) {
+         $input['ip_protocol'] = 'https';
+      }
+
```

Comment:
As discussed privately, you also will be able to set the port and you have to check set ports for ssl httpd-plugin before telling we should use `https`.
```suggestion
      if (!isset($metadata['httpd-port'])) {
         $input['port'] = $metadata['httpd-port'];
      }

      $input['ip_protocol'] = 'http';
      if (isset($metadata['httpd-plugins']['ssl']['ports'])) {
         $ssl_ports[] = $metadata['httpd-plugins']['ssl']['ports'];
         if (in_array("0", $ssl_ports) || (isset($metadata['httpd-port']) && in_array($metadata['httpd-port'], $ssl_ports))) {
            $input['ip_protocol'] = 'https';
         }
      }

```
The right logic is there but this is a suggestion and I didn't test it, so try it and fix it before merging.

This code is also following glpi agent commit: https://github.com/glpi-project/glpi-agent/commit/209f730be0f98797ce7bb5fa10ea63ee2da6339b
Here is an example of what you can obtain from [glpi-agent](https://github.com/glpi-project/glpi-agent) during CONTACT request:
```
{
   "action": "contact",
   "deviceid": "asus-2021-01-26-17-04-22",
   "enabled-tasks": [
      "inventory"
   ],
   "httpd-plugins": {
      "inventory": "disabled",
      "proxy": {
         "disabled": "no",
         "glpi_protocol": "yes",
         "local_store": "/srv/glpi-agent/proxy",
         "max_pass_through": 5,
         "max_proxy_threads": 10,
         "maxrate": 30,
         "maxrate_period": 3600,
         "no_category": "",
         "only_local_store": 0,
         "port": "8081",
         "prolog_freq": 24,
         "url_path": "/proxy"
      },
      "secondaryproxy": "disabled",
      "ssl": {
         "disabled": "no",
         "ports": [
            "0",
            "8081"
         ],
         "ssl_cert_file": "cert.pem",
         "ssl_key_file": "key.pem"
      },
      "test": "disabled",
      "toolbox": "disabled"
   },
   "httpd-port": "62354",
   "installed-tasks": [
      "collect",
      "deploy",
      "esx",
      "inventory",
      "netdiscovery",
      "netinventory",
      "wakeonlan"
   ],
   "name": "GLPI-Agent",
   "version": "1.0-dev"
}
```